packages/phishing-controller/src/PhishingController.ts

```diff
@@ -861,6 +909,154 @@ export class PhishingController extends BaseController<
+    '0x79a': 'soneium-minato',
+    '0x14a34': 'base-sepolia',
+    '0xab5': 'abstract',
+    '0x849ea': 'zero-network',
+    '0x138de': 'berachain',
+    '0x82': 'unichain',
+    '0x7e4': 'ronin',
+    '0x127': 'hedera',
+  };
+
+  /**
+   * Scan multiple tokens for malicious activity in bulk.
+   *
+   * @param request - The bulk scan request containing chainId and tokens.
+   * @param request.chainId - The chain ID in hex format (e.g., '0x1' for Ethereum).
+   * @param request.tokens - Array of token addresses to scan.
+   * @returns A mapping of lowercase token addresses to their scan results. Tokens that fail to scan are omitted.
+   */
+  bulkScanTokens = async (
+    request: BulkTokenScanRequest,
+  ): Promise<BulkTokenScanResponse> => {
+    const { chainId, tokens } = request;
+
+    if (!tokens || tokens.length === 0) {
+      return {};
+    }
+
+    // Limit to 100 tokens per request
+    const MAX_TOKENS_PER_REQUEST = 100;
+    if (tokens.length > MAX_TOKENS_PER_REQUEST) {
+      console.warn(
+        `Maximum of ${MAX_TOKENS_PER_REQUEST} tokens allowed per request`,
+      );
+      return {};
+    }
+
+    const normalizedChainId = chainId.toLowerCase();
+    const chain = this.#chainIdToName[normalizedChainId];
+
+    if (!chain) {
+      console.warn(`Unknown chain ID: ${chainId}`);
+      return {};
+    }
+
+    const results: Record<string, TokenScanResult> = {};
+    const tokensToFetch: string[] = [];
+
+    // Check cache for each token
+    for (const tokenAddress of tokens) {
+      const normalizedAddress = tokenAddress.toLowerCase();
+      const cacheKey = `${normalizedChainId}:${normalizedAddress}`;
+      const cachedResult = this.#tokenScanCache.get(cacheKey);
+
+      if (cachedResult) {
+        results[normalizedAddress] = {
+          result_type: cachedResult.result_type,
+          chain: normalizedChainId,
+          address: normalizedAddress,
+        };
+      } else {
+        tokensToFetch.push(tokenAddress);
+      }
+    }
+
+    // If there are tokens to fetch, call the bulk token scan API
+    if (tokensToFetch.length > 0) {
+      try {
+        const apiResponse = await safelyExecuteWithTimeout(
+          async () => {
+            const res = await fetch(
+              `${SECURITY_ALERTS_BASE_URL}${TOKEN_BULK_SCANNING_ENDPOINT}`,
+              {
+                method: 'POST',
+                headers: {
+                  Accept: 'application/json',
+                  'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                  chain,
+                  tokens: tokensToFetch,
+                }),
+              },
+            );
+
+            if (!res.ok) {
+              console.warn(
+                `Token bulk screening API error: ${res.status} ${res.statusText}`,
+              );
+              return null;
+            }
+
+            return await res.json();
+          },
+          true,
+          8000, // 8 second timeout
```

Comment:
should this be a const, instead of adding a "magic number" and then a comment to make it readable? (calling it magic number because of the [`no-magic-numbers` eslint rule](https://eslint.org/docs/latest/rules/no-magic-numbers))